java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -181,12 +182,34 @@ public void visitAssignmentExpression(AssignmentExpressionTree tree) {
   }
 
   private static boolean usesAnnotation(ClassTree classTree, List<String> annotations) {
+    return usesAnnotation(classTree, annotations, !CHECK_LOCAL_NAME);
+  }
+
+  private static final boolean CHECK_LOCAL_NAME = true;
+
+  private static boolean usesAnnotation(ClassTree classTree, List<String> annotations, boolean checkLocalName) {
```

Comment:
Nitpick: because we have the notion of checks all over the codebase, I think it would be wiser to rename this parameter to be:
1. More indicative of the context it will be used in
2. More indicative of its use within the method
```suggestion
  private static boolean usesAnnotation(ClassTree classTree, List<String> annotations, boolean shouldCheckAnnotationLocalName) {
```